src/rustllvm/PassWrapper.cpp

```diff
@@ -97,6 +97,55 @@ LLVMRustAddPass(LLVMPassManagerRef PM, Pass *pass) {
     pm->add(pass);
 }
 
+#define GEN_SUBTARGETS \
+    SUBTARGET(X86) \
+    SUBTARGET(ARM) \
+    SUBTARGET(AArch64) \
+    SUBTARGET(Mips) \
+    SUBTARGET(PPC)
+
+#define SUBTARGET(x) namespace llvm {                \
+    extern const SubtargetFeatureKV x##FeatureKV[];  \
+    extern const SubtargetFeatureKV x##SubTypeKV[];  \
```

Comment:
Is this just reaching into the internals of LLVM without including header files? That seems... brittle?